### ISSUE TITLE
`bolt plan new` should default to use the current project name instead of failing

### ISSUE THREAD

**Issue by op-ct** [puppetlabs/bolt#2757]
## Describe the Bug

`bolt plan new` fails unless the plan's first name segment matches the project name.  

However: Bolt already knows the project name, so it should just add it if only one name segment is given.

## Steps to Reproduce

1. Run `bolt plan new foo --pp` from inside a project directory ("boltybolt"):

```console
$ bolt plan new foo --pp
First segment of plan name 'foo' must match project name 'boltybolt'. Did you mean 'boltybolt::foo'?
```

## Expected Behavior

```console
$ bolt plan new foo --pp
Created plan 'boltybolt::foo' at '/path/to/boltybolt/plans/foo.pp'

Show this plan with:
    bolt plan show boltybolt::foo
Run this plan with:
    bolt plan run boltybolt::foo
```

## Environment
 - Version: 3.5.0
 - Platform: Debian 10

**Comment by lucywyman** [comment]
Hm, I think the team agrees that the happy path here would be a better experience, but it introduces some cases where Bolt has to guess what the user meant (for example, in a project named "project" if the user runs "bolt plan new project::foo" do they want to create plans/foo.yaml or plans/project/foo.yaml?). Forcing users to include the project namespace also reinforces that the `namespace::planname` is the name of the plan (for example for `plan show` or `plan run`).

**Comment by hestonhoffman** [comment]
How about:
> Incomplete plan name: Plan names consist of the project or module name followed by the plan name. Did you mean: {project}::{name}

**Comment by beechtom** [comment]
Closed by #2779
